src/lib/components/tooltip/QTooltip.svelte

```diff
@@ -1,17 +1,183 @@
-<script lang="ts">
+<script lang="ts" generics="T extends HTMLElement | string">
+  import { mount, unmount, onMount, untrack } from "svelte";
+  import QTooltipBase from "./QTooltipBase.svelte";
   import type { QTooltipProps } from "./props";
 
-  let { value, position = "bottom", children, ...props }: QTooltipProps = $props();
+  let {
+    target,
+    value = $bindable(false),
+    position = "bottom",
+    offset = { x: 0, y: 0 },
+    delay = 50,
+    hideDelay = 50,
+    children,
+    ...props
+  }: QTooltipProps<T> = $props();
 
-  Q.classes("q-tooltip", {
-    bemClasses: {
-      active: value,
-      [position]: true,
-    },
-    classes: [props.class],
+  let tooltipHelperEl = $state<HTMLDivElement>();
+  let tooltipEl = $state<HTMLDivElement>();
+
+  let realTarget = $state<HTMLElement>();
+
+  let timerShow = $state<ReturnType<typeof setTimeout> | null>(null);
+  let timerHide = $state<ReturnType<typeof setTimeout> | null>(null);
+
+  let mountedTooltip: ReturnType<typeof mountTooltip> | null = null;
+
+  const id = Date.now();
+
+  $effect(() => {
+    if (value) {
```

Comment:
Could this be done without the `if` and a `ternary` instead or would it loose reactivity then?

Reply:
Wdym "and a ternary"? You mean writing `value ? untrack(show) : untrack(hide)`? Should be doable

Reply:
Done, doesn't break anything